src/sgl/device/query.cpp

```
@@ -26,13 +26,27 @@ void QueryPool::reset()
     SLANG_RHI_CALL(m_rhi_query_pool->reset(), m_device);
 }
 
+void QueryPool::reset(uint32_t index, uint32_t count)
+{
+    SGL_CHECK(index + count <= m_desc.count, "'index' / 'count' out of range");
+    SLANG_RHI_CALL(m_rhi_query_pool->reset(index, count), m_device);
+}
+
 void QueryPool::get_results(uint32_t index, uint32_t count, std::span<uint64_t> result)
 {
     SGL_CHECK(index + count <= m_desc.count, "'index' / 'count' out of range");
```

Cppcheck (2.20.0)

[style] The function 'class_name' is never used.

(unusedFunction)